### DISCUSSION TITLE
I keep getting `Error: No QueryClient set, use QueryClientProvider to set one` on Sentry

### DISCUSSION THREAD

**Discussion by goenning** [TanStack/query#8145] [Q&A]
I keep getting these `Error: No QueryClient set, use QueryClientProvider to set one` from Sentry. It's not always though, happens occasionally like once a day, always with a different user. QueryClientProvider is the top most provider on my app, and I really don't know why this could be happening.

There's no way for me to reproduce this, and it never happened to me personally. Just wondering if anyone has seen this as well?

```ts
import { createRoot } from "react-dom/client";
import { StrictMode } from "react";
import { RouterProvider } from "react-router-dom";
import { router } from "./router";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { SentryInstrument } from "./SentryInstrument";

const root = document.getElementById("root")!;

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retryDelay: 1000,
      refetchOnWindowFocus: false,
      refetchOnMount: false,
    },
  },
});

createRoot(root).render(
  <StrictMode>
    <QueryClientProvider client={queryClient}>
        <RouterProvider router={router} />
      <SentryInstrument />
    </QueryClientProvider>
  </StrictMode>
);
```

**Comment by SaadNiaziDev** [comment]
[Tanstack + react-router-dom](https://tanstack.com/query/latest/docs/framework/react/examples/react-router
)
 I also encountered this issue if i use above example and initializing queryClient in main.jsx and routing in routes.jsx.
If i put all in same file, it works but if i keep them seperate and pass through props it gives same error once in a while.

Even if i do it the exact way as tanstack example does, it appears sometimes but since i turned strict mode off, no error yets
